docs/del3/Ytelse.md

```diff
@@ -1 +1,166 @@
+  minutter. Det kan ta noen minutter før man får utslag i grafene, så bare start
+  kommandoen og gå videre til punktenen under.
+- Gå deretter til DynamoDB-tabellen for applikasjonen din i AWS-konsollet og gå
+  til fanen _Metrics_. Her kan du se aktiviteten for tabellen i forskjellige
+  grafer. Blant annet ser man antall lesninger og skrivinger i sekundet og
+  ventetid (latency) for lesing og skriving mm. Det er desverre ikke noen
+  automatisk oppdatering av grafene så du må selv trykke på oppdateringskappen
+  oppe til høyret i Metrics vinduet.
+- Sett _Time Range_ oppe til høyre på siden til _Last Hour_ for å se de nyeste
+  hendelsene. Når ytelsestesten har kjørt noen minutter vil man kunne begynne å
+  se utslag på grafene. Prøv med forskjellige antall spørringer i sekundet og se
+  hvordan det påvirker ytelsen til databasen.
+- Alt etter hvordan du konfigurerte DynamoDB-tabellen i de tidligere oppgavene
+  vil du nå et punkt hvor spørringer blir blokkert. Disse vil du se i grafene
+  «Throttled read requests» og «Throttled write requests».
+
+>Hvis man har satt «Provisioned read capacity units» til 1 vil man kunne gjøre
+én lesing i sekundet (pluss en viss mengde over i kortere perioder). Hvis man
+utfører en test med 5 spørringer i sekundet vil man da være over grensen og om
+man kjører testen over en litt forlenget periode vil man se at
+spørringer over den konfigurerte grensen blir blokkert.
+
+
+## Oppsett av auto-scaling for Dynamodb
+
+Det er mulig å sette opp auto-skalering av DynamoDB-tabellene med noen få steg:
+
+- Gå til DynamoDB-tabellen for applikasjonen og velg fanen _Capacity_
+- Under seksjonen _Auto Scaling_ finner du muligheten for å konfigurere auto-
+  skalering
+- Huk av sjekkboksene for _Read capacity_ og _Write capacity_
+- Under _IAM Role_ velger du _Existing role with pre-defined policies_ og fyller
+inn rollen TODO: [navn ferdigoppsatt rolle her].
+
+Du kan nå kjøre ytelsestesten på 40 spørringer i sekundet igjen og følge med i
+Metrics-fanen for DynamoDB-tabellen. Du skal nå kunne se at den røde linjen for
+_Read capacity_-grafen justerer seg etter trykket av ytelsestesten.
+
+Du har nå en applikasjon som skalerer helt automatisk!
+
+
+## Oppsett av monitorering og konfigurering av struping for API gateway
+
+API Gateway har som standard en begrensning på 1000 spørringer i sekundet og
+utbruddsgrense (burst) på 5000 spørringer i sekundet i kortere perioder.
+
+Disse grensene har du kanskje ikke nådd i testingen din, men i en
+produksjonssetting kan det være være praktisk å begrense spørringer på API-nivå.
+Det er mulig å konfigurere disse grensene både høyere og lavere, alt etter
+behov.
+
+Konfigurasjon av dette gjøres under _Stages_-menyen API-et ditt:
+
+- Gå til _Stages_ til API-et for applikasjonen
+- Under Settings-fanen finner du en seksjon som heter _Default Method Throttling_
+- Her kan man skru av struping (throttlig) eller endre verdiene for strupingen
```

Comment:
Hva med å sette dette til et lavt tall, og så kjøre en ytelsestest for å faktisk se at requester blir strupet?

Reply:
Ja det kan jo være interessant å prøve ut! 😄